monkestation/code/modules/cybernetics/implant_items/weapons/mantis_blade.dm

```diff
@@ -96,3 +82,76 @@
 		living.stamina?.adjust(-30) // cost of a lunge
 
 	attack(target, user)
+
+
+
+
+/////////SHIELD MANTIS BLADES/////////////////
+/obj/item/mantis_blade/modified
+	name = "Modified C.H.R.O.M.A.T.A. mantis blade"
+	desc = "Modified mantis blades with bigger and wider blades, allowing user to block incoming projectiles and attacks. Because of that, the edge of blades is rather dull and large which makes it worse at wounding and requires much more time between each slash."
+	icon_state = "shield_mantis"
+	inhand_icon_state = "shield_mantis"
+	lefthand_file = 'monkestation/code/modules/cybernetics/icons/swords_lefthand.dmi'
+	righthand_file = 'monkestation/code/modules/cybernetics/icons/swords_righthand.dmi'
+	force = 10
+	wound_bonus = 10
+	attack_speed = 12
+	var/in_stance = FALSE
+
+/obj/item/mantis_blade/modified/attack_self(mob/user)
+	if (!isliving(user))
+		return
+	var/mob/living/owner = user
+	if (!in_stance)
+		var/obj/item/r_hand = owner.get_held_items_for_side(RIGHT_HANDS, FALSE)
+		var/obj/item/l_hand = owner.get_held_items_for_side(LEFT_HANDS, FALSE)
+		if(!istype(l_hand, r_hand))
+			to_chat(owner, span_warning("You must dual wield blades to enter the stance."))
+			return
+		owner.apply_status_effect(/datum/status_effect/shield_mantis_defense)
+		in_stance = TRUE
+		to_chat(owner, span_notice("You enter defensive stance with your mantis blades."))
+		return
+	owner.remove_status_effect(/datum/status_effect/shield_mantis_defense)
+	in_stance = FALSE
+	to_chat(owner, span_notice("You stop blocking with your blades."))
+
+
+/datum/status_effect/shield_mantis_defense
+	id = "mantis_defensive"
+	alert_type = /atom/movable/screen/alert/status_effect/realignment
+	//storing held items for when it was applied
+	var/obj/item/mantis_blade/modified/r_hand = null
+	var/obj/item/mantis_blade/modified/l_hand = null
+
+/datum/status_effect/shield_mantis_defense/on_apply()
+	. = ..()
+	r_hand = owner.get_held_items_for_side(RIGHT_HANDS, FALSE)
+	l_hand = owner.get_held_items_for_side(LEFT_HANDS, FALSE)
+	r_hand.block_chance += 30
+	l_hand.block_chance += 30
+	ADD_TRAIT(owner, TRAIT_CANT_ATTACK, id)
+	owner.add_movespeed_modifier(/datum/movespeed_modifier/shield_blades)
+
+/datum/status_effect/shield_mantis_defense/on_remove()
+	. = ..()
+	r_hand.block_chance = initial(r_hand.block_chance)
+	l_hand.block_chance = initial(l_hand.block_chance)
+	REMOVE_TRAIT(owner, TRAIT_CANT_ATTACK, id)
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/shield_blades)
+
+/datum/status_effect/shield_mantis_defense/tick()
+	. = ..()
+	var/held_r = owner.get_held_items_for_side(RIGHT_HANDS, FALSE)
+	var/held_l = owner.get_held_items_for_side(LEFT_HANDS, FALSE)
+	if(!istype(r_hand, held_r) || !istype(l_hand, held_l))
```

Comment:
why not just do this check on dropped() and destroy() for the shields? just make it so when its dropped it checks for an active defense effect and removes it when detected. Should cause a lot less overhead (even if its negligible)